manifests/plugin/dynflow.pp

```diff
@@ -49,36 +49,40 @@
   }
 
   if $external_core {
-    foreman_proxy::plugin { 'dynflow_core':
-      notify  => Service['smart_proxy_dynflow_core'],
-    }
+    $service = 'smart_proxy_dynflow_core'
 
     file { '/etc/smart_proxy_dynflow_core/settings.yml':
       ensure  => file,
       content => template('foreman_proxy/plugin/dynflow_core.yml.erb'),
       require => Foreman_proxy::Plugin['dynflow_core'],
-      notify  => Service['smart_proxy_dynflow_core'],
+      notify  => Service[$service],
     }
 
     file { '/etc/smart_proxy_dynflow_core/settings.d':
       ensure  => link,
       target  => "${foreman_proxy::config_dir}/settings.d",
       require => Foreman_proxy::Plugin['dynflow_core'],
-      notify  => Service['smart_proxy_dynflow_core'],
+      notify  => Service[$service],
     }
+  } else {
+    $service = 'foreman-proxy'
+  }
 
-    systemd::service_limits { 'smart_proxy_dynflow_core.service':
-      limits          => {
-        'LimitNOFILE' => $open_file_limit,
-      },
-      restart_service => false,
-      require         => Foreman_proxy::Plugin['dynflow_core'],
-      notify          => Service['smart_proxy_dynflow_core'],
-    }
+  foreman_proxy::plugin { 'dynflow_core':
```

Comment:
Why is the `dynflow_core` plugin now always needed? I kind of thought it wouldn't be needed at all anymore (matching the Debian deployment).

Reply:
The code from it is always needed when you want dynflow on the proxy, it just depends whether it is loaded into smart proxy or if it runs as a standalone process. Even if this wasn't included here, it would still get pulled in as a dependency of regular smart-proxy-dynflow, however being explicit about it felt right.

Reply:
Ok, that makes sense. I like being explicit.